src/easy/windows.rs

```diff
@@ -0,0 +1,139 @@
+#![allow(non_camel_case_types, non_snake_case)]
+
+use libc::c_void;
+
+#[cfg(target_env = "msvc")]
+mod win {
+    use curl_sys;
+    use kernel32;
+    use libc::{c_int, c_long, c_uchar, c_void};
+    use std::ffi::{CStr, CString};
+    use std::mem;
+    use std::ptr;
+    use schannel::cert_context::ValidUses;
+    use schannel::cert_store::CertStore;
+    use winapi;
+
+    fn lookup(module: &str, symbol: &str) -> Option<*const ::std::os::raw::c_void> {
+        unsafe {
+            let symbol = CString::new(symbol).unwrap();
+            let mut mod_buf: Vec<u16> = module.encode_utf16().collect();
+            mod_buf.push(0);
+            let handle = kernel32::GetModuleHandleW(mod_buf.as_mut_ptr());
+            let n = kernel32::GetProcAddress(handle, symbol.as_ptr());
+            if n == ptr::null() {
+                None
+            } else {
+                Some(n)
+            }
+        }
+    }
+
+    pub enum X509_STORE {}
+    pub enum X509 {}
+    pub enum SSL_CTX {}
+
+    type d2i_X509_fn = unsafe extern "C" fn(
+        a: *mut *mut X509,
+        pp: *mut *const c_uchar,
+        length: c_long,
+    ) -> *mut X509;
+    type X509_free_fn = unsafe extern "C" fn(x: *mut X509);
+    type X509_STORE_add_cert_fn = unsafe extern "C" fn(store: *mut X509_STORE, x: *mut X509)
+        -> c_int;
+    type SSL_CTX_get_cert_store_fn = unsafe extern "C" fn(ctx: *const SSL_CTX)
+        -> *mut X509_STORE;
+
+    struct OpenSSL {
+        d2i_X509: d2i_X509_fn,
+        X509_free: X509_free_fn,
+        X509_STORE_add_cert: X509_STORE_add_cert_fn,
+        SSL_CTX_get_cert_store: SSL_CTX_get_cert_store_fn,
+    }
+
+    fn lookup_functions(crypto_module: &str, ssl_module: &str) -> Option<OpenSSL> {
+        let d2i_X509 = lookup(crypto_module, "d2i_X509");
+        let X509_free = lookup(crypto_module, "X509_free");
+        let X509_STORE_add_cert = lookup(crypto_module, "X509_STORE_add_cert");
+        let SSL_CTX_get_cert_store = lookup(ssl_module, "SSL_CTX_get_cert_store");
+
+        if d2i_X509.is_some() && X509_free.is_some() && X509_STORE_add_cert.is_some() &&
+            SSL_CTX_get_cert_store.is_some()
+        {
+            unsafe {
+                Some(OpenSSL {
+                    d2i_X509: mem::transmute(d2i_X509.unwrap()),
+                    X509_free: mem::transmute(X509_free.unwrap()),
+                    X509_STORE_add_cert: mem::transmute(X509_STORE_add_cert.unwrap()),
+                    SSL_CTX_get_cert_store: mem::transmute(SSL_CTX_get_cert_store.unwrap()),
+                })
+            }
+        } else {
+            None
+        }
+    }
+
+    pub fn add_certs_to_context(ssl_ctx: *mut c_void) {
+        unsafe {
+            // check the runtime version of OpenSSL
+            let curl_ver = curl_sys::curl_version_info(curl_sys::CURLVERSION_NOW);
```

Comment:
Could this use the [`Version` bindings](https://docs.rs/curl/0.4.8/curl/struct.Version.html) to avoid raw FFI?